### DISCUSSION TITLE
Hash/octothorpe in db password

### DISCUSSION THREAD

**Discussion by allenluce** [amacneil/dbmate#413] [Support] [answered]
Issue #289 mentions that a + in a password can be escaped to allow it but that doesn't seem to be the case with #. When attempting to use a password (set in DATABASE_URL) with a #, I get an error:

```
% export DATABASE_URL='postgres://database_user:1HBwuj1RTknE2A3Jt2sxmGj6ltC#SGJJl@database_host:9382/dbname'
% dbmate dump
Error: parse "postgres://database_user:1HBwuj1RTknE2A3Jt2sxmGj6ltC": invalid port ":1HBwuj1RTknE2A3Jt2sxmGj6ltC" after host
```

Escaping doesn't seem to help:
```
% export DATABASE_URL='postgres://database_user:1HBwuj1RTknE2A3Jt2sxmGj6ltC\#SGJJl@database_host:9382/dbname'
% dbmate dump
Error: parse "postgres://database_user:1HBwuj1RTknE2A3Jt2sxmGj6ltC\\": invalid port ":1HBwuj1RTknE2A3Jt2sxmGj6ltC\\" after host
```

**Comment by amacneil** [comment, marked as answer]
In https://github.com/amacneil/dbmate/issues/289 the escaping is referring to their shell.

In your case I think you need to correctly URL encode your password. For example: `1HBwuj1RTknE2A3Jt2sxmGj6ltC%23SGJJl`

**Reply by allenluce** [reply]
Yup, that totally does it.
